docs/proposals/1374-multi-cluster-inference/README.md

```diff
@@ -0,0 +1,407 @@
+an [InferencePool](https://gateway-api-inference-extension.sigs.k8s.io/api-types/inferencepool/) and routing the request to a backend model server
+based on the EPP-provided endpoint. Although other multi-cluster inference approaches may exist, this proposal extends the current model to support
+multi-cluster routing so capacity in one cluster can serve traffic originating in another cluster or outside the cluster.
+
+### Why Multi-Cluster?
+
+GPU capacity is scarce and fragmented. Many users operate multiple clusters across regions and providers. A single cluster rarely satisfies peak or
+sustained demand, so a prescribed approach is required to share GPU capacity across clusters by:
+
+- Exporting an InferencePool from a source (“exporting”) cluster.
+- Importing the exported InferencePool into one or more destination (“importing”) clusters with enough detail for IGs to route requests to the associated
+  remote model server Pods.
+
+### Goals
+
+- Enable IGs to route to a group of common model server Pods, e.g. InferencePools, that exist in different clusters.
+- Align the UX with familiar [Multi-Cluster Services (MCS)](https://multicluster.sigs.k8s.io/concepts/multicluster-services-api/) concepts (export/import).
+- Keep the API simple and implementation-agnostic.
+
+### Non-Goals
+
+- Managing DNS or automatic naming.
+- Over-specifying implementation details to satisfy a single approach to multi-cluster inference.
+
+## Design Proposal
+
+The Multi-Cluster Inference Pooling (MCIP) model will largely follow the Multi-Cluster Services (MCS) model, with a few key differences:
+
+- DNS and ClusterIP resolution will be omitted, e.g. ClusterSetIP.
+- A separate export resource will be avoided, e.g. ServiceExport, by inlining the concept within InferencePool.
+
+An InferencePoolImport resource is introduced that is meant to be fully managed by a controller. This resource provides the information
+required for IGs to route LLM requests to model server endpoints of an InferencePool in remote clusters. How the IG routes the request to the remote
+cluster is implementation-specific.
+
+### Routing Modes
+
+The proposal supports the following routing modes:
+
+- Endpoint Mode: An IG of an importing cluster routes to endpoints selected by the EPP of the exported InferencePool. Pod and Service network connectivity
+  MUST exist between cluster members.
+- Parent Mode: An IG of an importing cluster routes to parents, e.g. Gateways, of the exported InferencePool. Parent connectivity MUST exist between cluster
+  members.
+
+### Workflow
+
+1. **Export an InferencePool:** An [Inference Platform Owner](https://gateway-api-inference-extension.sigs.k8s.io/concepts/roles-and-personas/)
+   exports an InferencePool by annotating it.
+2. **Exporting Controller:**
+   - Watches exported InferencePool resources (must have access to the K8s API server).
```

Comment:
1. if this controller needs to watch all the inferencePool resources, sig MC has a KEP to avoid storing secrete in the controller https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/5339-clusterprofile-plugin-credentials
2. Most of the multi-cluster manager projects (Kubefleet/OCM etc) uses pull mode instead of push so they don't watch any of the resources in the working cluster